KeenClient/KeenClient.m

```diff
@@ -354,6 +354,11 @@ + (KeenClient *)sharedClientWithProjectID:(NSString *)projectID
         return nil;
     }
 
+    // If the keys are already set, return the existing instance, don't overwrite the keys
+    if (self.sharedClient.projectID || self.sharedClient.writeKey || self.sharedClient.readKey) {
+        return self.sharedClient;
```

Comment:
One other thought on this... maybe we could add a call KCLogError here to make it clear something went wrong. I can imagine someone banging their head against the wall if they actually try to do this and the keys aren't updated. Maybe also return nil?